### PR TITLE
heuristic callback for BC

### LINKED ISSUE #30: give feasible solutions to solver
When running the BC algorithm, pajarito sometimes finds feasible solution. These should be given back to the solver by using a heuristic callback.

### PR DESCRIPTION
**Pull request by emreyamangil**
This PR adds a heuristic callback to BC algorithm.

Closes #30

### CODE REVIEW

File: src/nonlinear.jl [hunk 1]
```diff
@@ -627,8 +631,18 @@ function MathProgBase.optimize!(m::PajaritoModel)
         (cycle_indicator && m.status != :Optimal) && warn("Mixed-integer cycling detected, terminating Pajarito...")
     end
 
+    function heuristiccallback(cb)
+        if nlp_status == :Optimal
+            for i = 1:m.numVar
+                setSolutionValue!(cb, m.mip_x[i], nlp_solution[i])
```
Comment (mlubin) [inline src/nonlinear.jl]: What about the socp disaggregator variables?

Comment (emreyamangil) [inline src/nonlinear.jl]: Done!! (only for conic though as we don't do disaggregator for nonlinear)

**Comment by mlubin** [comment]
The SOCP disaggregator code isn't covered by tests. We should have a test that uses GLPK with BC to solve an MISOCP with disaggregator on.

**Comment by emreyamangil** [comment]
On it!

File: src/conic.jl [hunk 2]
```diff
@@ -777,6 +777,27 @@ function compareIntegerSolutions(m::PajaritoConicModel, sol1, sol2)
...
 
+function completeSOCPDisaggregator(m::PajaritoConicModel, solution)
+
+    new_solution = copy(solution)
+    if m.socp_disaggregator
+        k = 1
+        for (cone,ind) in m.pajarito_var_cones
+            if cone == :SOC
+                j = 1
+                for i in ind[2:end]
```
Comment (mlubin) [inline src/conic.jl]: What about the first variable in the cone?

File: src/conic.jl [hunk 3]
```diff
@@ -777,6 +777,27 @@ function compareIntegerSolutions(m::PajaritoConicModel, sol1, sol2)
     return round(sol1[int_ind]) == round(sol2[int_ind])
 end
 
+function completeSOCPDisaggregator(m::PajaritoConicModel, solution)
+
+    new_solution = copy(solution)
+    if m.socp_disaggregator
+        k = 1
```
Comment (mlubin) [inline src/conic.jl]: ``k`` and ``j`` aren't used?

File: src/conic.jl (hunk 2 above)
Comment (mlubin) [inline src/conic.jl]: Never mind, I get it

Comment (emreyamangil) [inline src/conic.jl]: We don't add a variable for that one, we are just disaggregating the squares.

File: src/conic.jl (hunk 3 above)
Comment (emreyamangil) [inline src/conic.jl]: Good point!! :D

File: src/conic.jl (hunk 2 above)
Comment (emreyamangil) [inline src/conic.jl]: You are faster than my typing skills :D

**Pull request merged**